x/mm2/types/provider.go

```diff
@@ -0,0 +1,66 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/skip-mev/slinky/pkg/json"
+)
+
+// ValidateBasic performs basic validation on a ProviderConfig.
+func (pc *ProviderConfig) ValidateBasic() error {
+	if len(pc.Name) == 0 {
+		return fmt.Errorf("provider name must not be empty")
+	}
+
+	if len(pc.OffChainTicker) == 0 {
+		return fmt.Errorf("provider offchain ticker must not be empty")
+	}
+
+	// NormalizeByPair is allowed to be empty
+	if pc.NormalizeByPair != nil {
+		if err := pc.NormalizeByPair.ValidateBasic(); err != nil {
+			return err
+		}
+	}
+
+	if len(pc.Metadata_JSON) > MaxMetadataJSONFieldLength {
+		return fmt.Errorf("metadata json field is longer than maximum length of %d", MaxMetadataJSONFieldLength)
+	}
+
+	if err := json.IsValid([]byte(pc.Metadata_JSON)); err != nil {
+		return fmt.Errorf("invalid provider config metadata json: %w", err)
+	}
```

Comment:
should we add any validation that ProviderConfig.metadata_JSON holds any particular data?

Reply:
We can at a later time when we have solid metadata types we care about.  Now that we are adding defi providers like raydium, they use this field, so we can add those once merged.  Good callout.  adding a linear issue for us to track this